nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/DocumentBrowsingTest.java

```diff
@@ -487,10 +489,85 @@ public void itCanSetArrayPropertyToNull() throws IOException {
+                Collections.singletonMap("X-NXDocumentProperties", "defaultvalue"))) {
+
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonNode jsonProperties = node.get("properties");
+            assertNotNull(jsonProperties);
+            JsonNode jsonMultiComplex = jsonProperties.get("dv:multiComplexWithoutDefault");
+            assertNotNull(jsonMultiComplex);
+            assertTrue(jsonMultiComplex.isArray());
+            assertEquals(0, jsonMultiComplex.size());
+
+            // Then the document is updated
+            fetchInvalidations();
+            doc = session.getDocument(doc.getRef());
+            // ListProperty returns an empty List whenever is empty or null
+            Serializable multiComplex = doc.getPropertyValue("dv:multiComplexWithoutDefault");
+            assertTrue(multiComplex instanceof List);
+            assertTrue(((List<?>) multiComplex).isEmpty());
+        }
+    }
+
+    /*
+     * NXP-28433
+     */
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.restapi.server:test-defaultvalue-docTypes.xml")
+    public void itCanSetArrayPropertyOfComplexToNull() throws IOException {
+        DocumentModel doc = session.createDocumentModel("/", "myDocument", "DocDefaultValue");
+        Map<String, String> complex = new HashMap<>();
+        complex.put("foo", "val1");
+        complex.put("bar", "val2");
+        doc.setProperty("defaultvalue", "multiComplexWithoutDefault", Collections.singletonList(complex));
+        doc = session.createDocument(doc);
+        fetchInvalidations();
+
+        try (CloseableClientResponse response = getResponse(RequestType.PUT, "id/" + doc.getId(),
+                "{\"entity-type\":\"document\",\"properties\":{\"dv:multiComplexWithoutDefault\":null}}",
+                Collections.singletonMap("X-NXDocumentProperties", "defaultvalue"))) {
+
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonNode jsonProperties = node.get("properties");
+            assertNotNull(jsonProperties);
+            JsonNode jsonMultiComplex = jsonProperties.get("dv:multiComplexWithoutDefault");
+            assertNotNull(jsonMultiComplex);
+            assertTrue(jsonMultiComplex.isArray());
+            assertEquals(0, jsonMultiComplex.size());
+
+            // Then the document is updated
+            fetchInvalidations();
+            doc = session.getDocument(doc.getRef());
+            // ListProperty returns an empty List whenever is empty or null
+            Serializable multiComplex = doc.getPropertyValue("dv:multiComplexWithoutDefault");
+            assertTrue(multiComplex instanceof List);
+            assertTrue(((List<?>) multiComplex).isEmpty());
+        }
+    }
+
```

Comment:
i am wondering what if we extract the main treatment of the two test as they check the same things the main difference is in a case we pass a `null` and an other `[]` but at the end we have an empty array.

Reply:
here we end with a list and not an array regarding the other test which is array. I don't see the point to have a method for that, I think it increases complexity with no real advantage and, in such case, I prefer to have the complet test in front of me when reading than having to navigate in the test code to find what it does.